src/main/java/at/hannibal2/skyhanni/features/inventory/wardrobe/CustomWardrobe.kt

```diff
@@ -0,0 +1,518 @@
+import at.hannibal2.skyhanni.features.inventory.wardrobe.WardrobeAPI.currentWardrobeSlot
+import at.hannibal2.skyhanni.features.inventory.wardrobe.WardrobeAPI.inCustomWardrobe
+import at.hannibal2.skyhanni.features.inventory.wardrobe.WardrobeAPI.inWardrobe
+import at.hannibal2.skyhanni.mixins.transformers.gui.AccessorGuiContainer
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.ColorUtils.darker
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColorInt
+import at.hannibal2.skyhanni.utils.ColorUtils.withAlpha
+import at.hannibal2.skyhanni.utils.ConfigUtils.jumpToEditor
+import at.hannibal2.skyhanni.utils.DelayedRun
+import at.hannibal2.skyhanni.utils.EntityUtils.getFakePlayer
+import at.hannibal2.skyhanni.utils.InventoryUtils.clickSlot
+import at.hannibal2.skyhanni.utils.InventoryUtils.getWindowId
+import at.hannibal2.skyhanni.utils.ItemUtils.removeEnchants
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.HorizontalAlignment
+import at.hannibal2.skyhanni.utils.RenderUtils.VerticalAlignment
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderable
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiContainer
+import net.minecraft.client.renderer.GlStateManager
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.math.max
+import kotlin.time.Duration.Companion.milliseconds
+
+@SkyHanniModule
+object CustomWardrobe {
+
+    private val config get() = SkyHanniMod.feature.inventory.customWardrobe
+
+    private var displayRenderable: Renderable? = null
+    private var buttonsRenderable: Renderable? = null
+    private var inventoryButton: Renderable? = null
+    private var editMode = false
+    private var waitingForInventoryUpdate = false
+
+    @SubscribeEvent
+    fun onGuiRender(event: GuiContainerEvent.BeforeDraw) {
+        if (!isEnabled() || editMode) return
+        event.cancel()
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent) {
+        if (!isEnabled()) return
+        val gui = Minecraft.getMinecraft().currentScreen as? GuiContainer ?: return
+        if (editMode) { // Inventory Button for re-enabling the Custom Wardrobe
+            val renderable = inventoryButton ?: addReEnableButton().also { inventoryButton = it }
+            val accessorGui = gui as AccessorGuiContainer
+            val posX = accessorGui.guiLeft + (1.05 * accessorGui.width).toInt()
+            val posY = accessorGui.guiTop + (accessorGui.height - renderable.height) / 2
+            Position(posX, posY).renderRenderable(renderable, posLabel = "Custom Wardrobe", addToGuiManager = false)
+            return
+        }
```

Comment:
This is a mess. Render the custom wadrobe in the BeforeDraw event and the editMode thing in GuiRenderEvent.ChestGuiOverlayEvent.